lib/segment/src/common/operation_time_statistics.rs

```diff
@@ -87,9 +138,39 @@ impl std::ops::Add for OperationDurationStatistics {
                 other.max_duration_micros,
                 |a, b| a > b,
             ),
+            total_duration_micros: self.total_duration_micros + other.total_duration_micros,
             last_responded: std::cmp::max(self.last_responded, other.last_responded),
+            duration_micros_histogram: merge_buckets(
+                self.duration_micros_histogram,
+                other.duration_micros_histogram,
+            ),
+        }
+    }
+}
+
+/// Merges two vectors of buckets, summing the counts of the same boundaries.
+/// # Panics
```

Comment:
This is a nitpick, but convention is to keep two empty lines around headers. I think it's better to follow that.

```suggestion
///
/// # Panics
///
```